File: Basic/15.Find the missing number in an array of size n containing numbers from 1 to n+1 in C/Solution.c
#include <stdio.h>

int findMissingNumber(int arr[], int n) {
    int total = (n + 1) * (n + 2) / 2; // Sum of 1 to n+1
    for (int i = 0; i < n; i++) {
        total -= arr[i]; // Subtract array elements from total
    }
    return total;
}

int main() {
    int n;
    printf("Enter the size of the array (n): ");
    scanf("%d", &n);

    int arr[n];
    printf("Enter %d elements of the array: ", n);
    for (int i = 0; i < n; i++) {
        scanf("%d", &arr[i]);
    }

    int missingNumber = findMissingNumber(arr, n);
    printf("The missing number is: %d\n", missingNumber);

    return 0;
}

File: Basic/15.Find the missing number in an array of size n containing numbers from 1 to n+1 in C/Solution.java
import java.util.Scanner;

public class MissingNumber {
    public static int findMissingNumber(int[] arr, int n) {
        int total = (n + 1) * (n + 2) / 2; // Sum of 1 to n+1
        for (int num : arr) {
            total -= num; // Subtract array elements from total
        }
        return total;
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        System.out.print("Enter the size of the array (n): ");
        int n = scanner.nextInt();

        int[] arr = new int[n];
        System.out.print("Enter " + n + " elements of the array: ");
        for (int i = 0; i < n; i++) {
            arr[i] = scanner.nextInt();
        }

        int missingNumber = findMissingNumber(arr, n);
        System.out.println("The missing number is: " + missingNumber);
    }
}
